frontend/src/hooks/messages.ts

```diff
@@ -26,6 +32,87 @@ export function getMessagesQuery({
   });
 }
 
+export function useSendMessageStreaming({
+  onAddResult,
+  onSettled,
+}: {
+  onAddResult: (result: IResultType) => void;
+  onSettled: () => void;
+}) {
+  const queryClient = useQueryClient();
+  const current_connection = useGetRelatedConnection();
+
+  return useMutation({
+    retry: false,
+    mutationFn: async ({
+      message,
+      conversationId,
+      execute = true,
+    }: {
+      message: string;
+      conversationId: string;
+      execute?: boolean;
+    }) => {
+      const messageOptions = await queryClient.fetchQuery(
+        getMessageOptions(current_connection?.id)
+      );
+      let queryOut: {
+        human_message: IMessageOut;
+        ai_message: IMessageWithResultsOut;
+      };
+      await api.streamingQuery({
+        conversationId,
+        query: message,
+        execute,
+        message_options: messageOptions,
+        onMessage(event, data) {
+          if (event === StreamingEventType.QUERY_OUT.valueOf()) {
+            queryOut = JSON.parse(data);
+          } else if (event === StreamingEventType.ADD_RESULT.valueOf()) {
+            onAddResult(JSON.parse(data));
+          }
+        },
+      });
+
+      // TODO: ugly error
+      return queryOut;
+    },
+    onSuccess: (data, variables) => {
+      queryClient.setQueryData(
+        getMessagesQuery({ conversationId: variables.conversationId }).queryKey,
```

Comment:
Maybe add some comments here on why we need to update this. I like to not have to think at all when reading code, even if that means over-commenting. Daily brainpower is a limited resource.